src/app/containers/PageHandlers/withHashChangeHandler/isHashChange.js

```diff
@@ -0,0 +1,11 @@
+import equals from 'ramda/src/equals';
+
+export default (
+  { location: { hash: prevHash, ...prevLocation }, ...prevProps },
+  { location: { hash: nextHash, ...nextLocation }, ...nextProps },
+) => {
+  const propsAreEqual = equals(prevProps, nextProps);
```

Comment:
Is there any potentially performance impact here? I think the page components receive the large `pageData` object that could result in an expensive deep equality check?

Reply:
Yeh that's good point. There could be